# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/Lowering.cpp
# Commit: bd8381646b47
# Full Hash: bd8381646b4734c3240dace68c3606be1cc38ba2
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-02 04:10:47
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.6: Add tls dependency for truncate i32. r=lth
#   
#   We generate builtin call for MTruncateToInt32 operation for floating points types,
#   so we need to add a tls dependency.
#   I inserted NYI for arm64 because Ion doesn't support arm64.
# ==============================================================================

diff -r 81e645d5ccfd -r bd8381646b47 js/src/jit/Lowering.cpp
--- a/js/src/jit/Lowering.cpp	Thu Oct 01 07:53:01 2020 +0000
+++ b/js/src/jit/Lowering.cpp	Thu Oct 01 11:24:30 2020 +0000
@@ -2381,6 +2381,16 @@
   }
 }
 
+void LIRGenerator::visitWasmBuiltinTruncateToInt32(
+    MWasmBuiltinTruncateToInt32* truncate) {
+  mozilla::DebugOnly<MDefinition*> opd = truncate->input();
+  MOZ_ASSERT(opd->type() == MIRType::Double || opd->type() == MIRType::Float32);
+
+  // May call into JS::ToInt32() on the slow OOL path.
+  gen->setNeedsStaticStackAlignment();
+  lowerWasmBuiltinTruncateToInt32(truncate);
+}
+
 void LIRGenerator::visitWasmBoxValue(MWasmBoxValue* ins) {
   LWasmBoxValue* lir = new (alloc()) LWasmBoxValue(useBox(ins->input()));
   define(lir, ins);